src/hera/workflows/script.py

```diff
@@ -31,6 +34,21 @@
 from hera.workflows.task import Task
 
 
+@runtime_checkable
+class ScriptConstructor(Protocol):
+    def generate_source(self, instance: "Script") -> str:
+        raise NotImplementedError
+
+    def transform_inputs(self, cls: Type["Script"], values: Any) -> Any:
+        return values
+
+    def transform_script_template(self, instance: "Script", script: _ModelScriptTemplate) -> _ModelScriptTemplate:
+        return script
+
+    def transform_template(self, instance: "Script", template: _ModelTemplate) -> _ModelTemplate:
+        return template
```

Comment:
This is the general interface anyone can implement to create their own script constructor.

Hera allows users to pass a custom script constructor of their own or a string constant which corresponds to hera only script constructor (inline and runner)